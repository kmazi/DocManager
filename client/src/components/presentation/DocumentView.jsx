import React from 'react';
import propTypes from 'prop-types';

import { DocumentPreview } from './DocumentPreview';
import Search from '../container/Search';

const DocumentView = ({ documents, shouldDisplay, read, deleteId,
  documentStatus, readDocument, history, deleteDocument, paginateDocument,
  documentAccess, currentPage, documentsCount, roleType, id }) => {
  const finalRender = (shouldDisplay) ?
    (
      <div>
        <Search />
        <DocumentPreview
          userDocuments={documents}
          read={read}
          deleteId={deleteId}
          readDocument={readDocument}
          history={history}
          deleteDocument={deleteDocument}
          paginateDocument={paginateDocument}
          documentAccess={documentAccess}
          currentPage={currentPage}
          documentsCount={documentsCount}
          roleType={roleType}
          userId={id}
        />
      </div>
    ) :
      (<div>
        <Search />
        <p id="status" className="center-align">{documentStatus}</p>
      </div>);
  /**
 * Renders the html elements on the browser
 * @return {object} Returns the html object to render
 */
  return (// Render documents if successsfully loaded from server
    // else show loading message
    <div id="doc-view-background" >
      {finalRender}
    </div>);
};

DocumentView.propTypes = {
  documentStatus: propTypes.string.isRequired,
  documents: propTypes.arrayOf(propTypes.shape()).isRequired,
  history: propTypes.shape({
    push: propTypes.func.isRequired,
  }).isRequired,
  paginateDocument: propTypes.func.isRequired,
  readDocument: propTypes.func.isRequired,
  deleteDocument: propTypes.func.isRequired,
  read: propTypes.number.isRequired,
  deleteId: propTypes.number.isRequired,
  shouldDisplay: propTypes.bool.isRequired,
  id: propTypes.number.isRequired,
  documentsCount: propTypes.number.isRequired,
  currentPage: propTypes.number.isRequired,
  documentAccess: propTypes.string.isRequired,
  roleType: propTypes.string.isRequired,
};

export default DocumentView;
