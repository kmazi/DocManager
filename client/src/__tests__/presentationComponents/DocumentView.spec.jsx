import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DocumentView from '../../components/presentation/DocumentView';

describe('The DocumentView component:', () => {
  const props = {
    documents: [{ title: 'jackson goes to school',
      body: 'jackson goes to church',
      id: 1 }],
    shouldDisplay: true,
    read: 1,
    deleteId: 1,
    documentStatus: 'No document found!',
    readDocument: jest.fn(() => Promise.resolve()),
    history: { push: jest.fn() },
    deleteDocument: jest.fn(() => Promise.resolve()),
    paginateDocument: jest.fn(() => Promise.resolve()),
    documentAccess: 'Admin',
    currentPage: 1,
    documentsCount: 1,
    roleType: 'Learning',
    id: 1,
  };
  test('should render the DocumentView component correctly', () => {
    const enzymeWrapper = shallow(<DocumentView {...props} />);
    const component = enzymeWrapper;
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  test(`should render an error message
  when shoulddisplay property is false`, () => {
    props.shouldDisplay = false;
    const enzymeWrapper = shallow(<DocumentView {...props} />);
    const component = enzymeWrapper;
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
