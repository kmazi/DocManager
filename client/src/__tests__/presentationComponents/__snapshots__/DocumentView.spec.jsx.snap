// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The DocumentView component: should render an error message
  when shoulddisplay property is false 1`] = `
<div
  id="doc-view-background"
>
  <div>
    <Connect(withRouter(Search)) />
    <p
      className="center-align"
      id="status"
    >
      No document found!
    </p>
  </div>
</div>
`;

exports[`The DocumentView component: should render the DocumentView component correctly 1`] = `
<div
  id="doc-view-background"
>
  <div>
    <Connect(withRouter(Search)) />
    <DocumentPreview
      currentPage={1}
      deleteDocument={[Function]}
      deleteId={1}
      documentAccess="Admin"
      documentsCount={1}
      history={
        Object {
          "push": [Function],
        }
      }
      paginateDocument={[Function]}
      read={1}
      readDocument={[Function]}
      roleType="Learning"
      userDocuments={
        Array [
          Object {
            "body": "jackson goes to church",
            "id": 1,
            "title": "jackson goes to school",
          },
        ]
      }
      userId={1}
    />
  </div>
</div>
`;
